### ISSUE TITLE
Changing/disabling keyboard shortcuts can freeze the entire deskop

### ISSUE THREAD

**Issue by trialism** [mate-desktop/mate-terminal#249]
#### Expected behaviour

Changing/disabling keyboard shortcuts working properly.

#### Actual behaviour

Changing/disabling keyboard shortcuts crashes the entire mate desktop after modifying the state of 2-4 shortcuts.

#### Steps to reproduce the behaviour

Modify shortcuts. After 2-4 modifications the desktop will freeze.

#### MATE general version

mate-session 1.20.0

#### Package version

1.20.0

#### Linux Distribution

Ubuntu 18.04

#### GDB Error

```
Thread 1 "mate-terminal" received signal SIGSEGV, Segmentation fault.
    from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
```

**Comment by raveit65** [comment]
I can't reproduce this with fedora 28.
Is this issue confirmed by ubuntu MATE maintainer and did you filed out a report at ubuntu bugreporting system?

**Comment by trialism** [comment]
No, why would I use the ubuntu bugreporting system? This issue appears in mate-terminal.

**Comment by vkareh** [comment]
@trialism - this is probably a downstream issue only on Ubuntu MATE, since it has its own keybindings, so you might want to file a bug in launchpad. Try running `apport-bug mate-terminal` as soon as the crash happens, and the Ubuntu MATE team will get a stack trace of your bug, which would be helpful.

That said, do you have the mate-hud running by any chance? Also, which keyboard shortcuts did you change, and to what?

**Comment by trialism** [comment]
@vkareh It doesn't matter which keyboard shortcut I change - after changing 2-3 shortcuts mate-terminal crashes and freezes my mate desktop. I don't use mate-hud.

**Comment by vkareh** [comment]
okay, I can reproduce easily. In fact, in reproducing it reminded me of the fact that I've seen this before.

* mate-terminal: #197
* Ubuntu MATE: https://bugs.launchpad.net/ubuntu/+source/mate-terminal/+bug/1667227
* gnome-terminal: https://gitlab.gnome.org/GNOME/gtk/issues/180

However, when trying to reproduce in gnome-terminal, I could not. This means that somewhere along the line, gnome-terminal probably fixed this issue and maybe we can backport it.

**Comment by vkareh** [comment]
Confirmed on both Ubuntu MATE 18.04 and 18.10.
Was not able to reproduce on Fedora 28, Linux Mint, Solus, OpenSUSE Tumbleweed... based on the old reports, it seems like it only affects Ubuntu and Arch.

**Comment by lukefromdc** [comment]
I just confirmed this to be present on Debian Unstable as well. Had to go to console and kill mate-terminal to save my session
